AudioCompression/demo/audioCompression/algorithm/SerializationStepDemo.java

```diff
@@ -0,0 +1,60 @@
+package audioCompression.algorithm;
+
+import java.nio.ByteBuffer;
+import java.util.Random;
+
+import audioCompression.algorithm.SerializationStep;
+import audioCompression.types.AudioByteBuffer;
+import audioCompression.types.CompressedAudioFile;
+
+public class SerializationStepDemo {
+
+	public static void main(String[] args) {
+		
+		// This will test the implementation of the Serialization step
+		// create an audio Byte buffer, pass into the forward step, it should create
+		// a file with the data, and then pass that back through the reverse stage,
+		// the data on each end should be identical
+		
+		int numEle = 1000;
+		int []data = new int[numEle];
+		
+		Random r = new Random(System.currentTimeMillis());
+		ByteBuffer buff = ByteBuffer.allocate(numEle * (Integer.SIZE/Byte.SIZE));
+		for (int i = 0; i < numEle; ++i) {
+			data[i] = r.nextInt();
+			buff.putInt(data[i]);
+		}
+		
+		int testIndex = 10;
+		// intentionally change the data at one point to verify the original data stream
+		if (data[testIndex] != 5) {
+			data[testIndex] = 5;
+		} else {
+			data[testIndex] = 6;
+		}
+				
+	//	System.out.println("buffer position after inserting " + String.valueOf(numEle) + " is at " + String.valueOf(buff.position()));
+		buff.rewind();
+	//	System.out.println("buffer position after inserting " + String.valueOf(numEle) + " is at " + String.valueOf(buff.position()));
+		
+		AudioByteBuffer aBuff = new AudioByteBuffer(buff);
+		SerializationStep step = new SerializationStep();
```

Comment:
Maybe here we could have something like

SerializationStep step = new SerializationStep(" -- filename here --");

instead of passing the filename through the entire pipeline?